core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -437,4 +430,62 @@ static HttpResponse ofFailed(Throwable cause) {
         subscribe(aggregator, executor, SubscriptionOption.WITH_POOLED_OBJECTS);
         return future;
     }
+
+    /**
+     * Returns a new {@link HttpResponseDuplicator} that duplicates multiple {@link HttpResponse}s which publish
+     * the same elements with this {@link HttpResponse}.
+     * Note that you cannot subscribe to this {@link HttpResponse} anymore after you call this method.
+     * To subscribe, call {@link HttpResponseDuplicator#duplicate()} from the returned
+     * {@link HttpResponseDuplicator}.
+     */
+    @Override
+    default HttpResponseDuplicator toDuplicator() {
+        return toDuplicator(Flags.defaultMaxResponseLength());
+    }
+
+    /**
+     * Returns a new {@link HttpResponseDuplicator} that duplicates multiple {@link HttpResponse}s which publish
+     * the same elements with this {@link HttpResponse}.
+     * Note that you cannot subscribe to this {@link HttpResponse} anymore after you call this method.
+     * To subscribe, call {@link HttpResponseDuplicator#duplicate()} from the returned
+     * {@link HttpResponseDuplicator}.
+     *
+     * @param executor the executor to duplicate
+     */
+    @Override
+    default HttpResponseDuplicator toDuplicator(EventExecutor executor) {
+        return toDuplicator(executor, Flags.defaultMaxResponseLength());
+    }
+
+    /**
+     * Returns a new {@link HttpResponseDuplicator} that duplicates multiple {@link HttpResponse}s which publish
+     * the same elements with this {@link HttpResponse}.
+     * Note that you cannot subscribe to this {@link HttpResponse} anymore after you call this method.
+     * To subscribe, call {@link HttpResponseDuplicator#duplicate()} from the returned
+     * {@link HttpResponseDuplicator}.
+     *
+     * @param maxResponseLength the maximum response length that the duplicator can hold in its buffer.
+     *                         {@link ContentTooLargeException} is raised if the length of the buffered
+     *                         {@link HttpData} is greater than this value.
+     */
+    default HttpResponseDuplicator toDuplicator(long maxResponseLength) {
```

Comment:
nit: How about narrowing down the maxSingalLength to `int` here than truncating the `maxResponseLength` in `StreamMessageProcessor` constructor?

Reply:
I used long because `Flags.defaultMaxResponseLength()` uses long. Do you want me to change that as well?

Reply:
`defaultMaxResponseLength` must be long.

Reply:
So it's good to keep it as it is? @ikhoon

Reply:
> I used long because `Flags.defaultMaxResponseLength()` uses long.

The `Flags.defaultMaxResponseLength()` enventually cast `long` to `int`. So I thought the following because it can expose stricter type.
```java
@Override
default HttpResponseDuplicator toDuplicator(EventExecutor executor, int maxResponseLength) 
   return new DefaultHttpResponseDuplicator(this, executor, maxResponseLength);
}

@Override
default HttpResponseDuplicator toDuplicator(EventExecutor executor) 
    int maxResponseLength = Flags.defaultMaxResponseLength() > Integer.MAX_VALUE ? Integer.MAX_VAUE : Flags.defaultMaxResponseLength();
    return toDuplicator(executor, maxResponseLength);
}
```

But I'm not strong here. Let keep it!